skaha/src/main/java/org/opencadc/skaha/session/SessionAction.java

```diff
@@ -374,10 +374,15 @@ public void streamPodLogs(String forUserID, String sessionID, OutputStream out)
     public Session getSession(String forUserID, String sessionID) throws Exception {
         List<Session> sessions = getSessions(forUserID, sessionID);
         if (sessions.size() >0) {
-            return sessions.get(0);
-        } else {
-            throw new ResourceNotFoundException("session " + sessionID + " not found");
-        }
+            for (Session session : sessions) {
+                // exclude 'desktop-app'
+                if (!SkahaAction.TYPE_DESKTOP_APP.equalsIgnoreCase(session.getType())) {
+                    return session;
```

Comment:
This isn't related to action=renew, so does it fix another separate issue?

Reply:
Yes, there was a bug that I introduced that given a desktop session ID, skaha returned the desktop session as well as the desktop apps. The code only picked the first item returned which turned out to be almost always a desktop-app. This fixed the bug. I mentioned the bug fix and you were okay with putting the fix in this release.

Reply:
Okay, yes, thanks for the reminder.  Do you know if this will have implications on the portal?